packages/nimble-blazor/NimbleBlazor/Components/NimbleTable.razor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace NimbleBlazor;
+
+/// <summary>
+/// A table component
+/// </summary>
+/// <typeparam name="TData">Represents the type for a row of data in the table (an element of the <see cref="Data"/>).</typeparam>
+/// <remarks>The type represented by <see cref="TData"/> should not have any hierarchy. All aspects that can be serialized
+/// should be at the top-level.</remarks>
+public partial class NimbleTable<TData> : ComponentBase
+{
+    private ElementReference _table;
+    internal static string SetTableDataMethodName = "NimbleBlazor.Table.setData";
+
+    [Inject]
+    private IJSRuntime? JSRuntime { get; set; }
+
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the data for the table.
+    /// </summary>
+    [Parameter]
+    public IEnumerable<TData> Data { get; set; } = Enumerable.Empty<TData>();
+
+    /// <summary>
+    /// Gets or sets a callback that's invoked when the data changes
+    /// </summary>
+    [Parameter]
+    public EventCallback<IEnumerable<TData>> DataChanged { get; set; }
+
+    [Parameter(CaptureUnmatchedValues = true)]
+    public IDictionary<string, object>? AdditionalAttributes { get; set; }
+
+    /// <exception cref="JsonException">Provided Data is unsupported</exception>
+    protected override Task OnAfterRenderAsync(bool firstRender)
```

Comment:
Note that the build is reporting some warnings here.

Do you know what's our policy is for warnings in our C# styleguide? In the ESLint ruleset we try to promote all warnings to errors to force us to fix them and not accumulate a bunch of noise in our build output. Do you know if these would block submission? If not should we configure our project to promote them to errors (or propose that change in the styleguide)?

Reply:
Yeah, this seems to be blocking submission (the build step fails), so I think we're okay.